### PR TITLE
main signs in subsign details page

### PR DESCRIPTION
**Pull request by bdevloed**
cf https://binnenland.atlassian.net/browse/GN-2707

### CODE REVIEW

**Comment by bdevloed** [comment]
In draft - to do: float the "add main sign" to the right

File: app/controllers/road-sign-concepts/road-sign-concept.js [hunk 1]
```diff
@@ -51,6 +52,35 @@ export default class RoadsignConceptsRoadsignConceptController extends Controlle
     this.model.roadSignConcept.save();
   }
 
+  @action
+  setMainSignCodeFilter(event) {
+    this.mainSignCodeFilter = event.target.value.trim();
+  }
+
+  // @action
+  // async addMainSign(mainSign) {
```
Comment (Windvis) [inline app/controllers/road-sign-concepts/road-sign-concept.js:61]: Don't forget to remove this :smile:

File: app/routes/road-sign-concepts/road-sign-concept.js [hunk 2]
```diff
@@ -18,6 +18,11 @@ export default class RoadsignConcept extends Route {
           size: 10000,
         },
       }),
+      allMainSigns: this.store.findAll('road-sign-concept').then((mainSign) => {
```
Comment (Windvis) [inline app/routes/road-sign-concepts/road-sign-concept.js:21]: Is this still needed?

File: app/controllers/road-sign-concepts/road-sign-concept.js [hunk 3]
```diff
@@ -51,6 +52,35 @@ export default class RoadsignConceptsRoadsignConceptController extends Controlle
...
+    mainSigns.pushObject(mainSign);
+    this.categoryRoadSigns.removeObject(mainSign);
+    this.model.roadSignConcept.save();
+  }
+
+  @action
+  async removeMainSign(mainSign) {
+    let mainSigns = await this.model.roadSignConcept.mainSigns;
+    mainSigns.removeObject(mainSign);
+    this.model.allMainSigns.pushObject(mainSign);
```
Comment (Windvis) [inline app/controllers/road-sign-concepts/road-sign-concept.js:80]: It should probably modify the `categoryRoadSigns` similar to the add?

File: app/templates/road-sign-concepts/road-sign-concept.hbs [hunk 4]
```diff
@@ -306,6 +397,69 @@
...
+              {{t "mainSign.attr.category"}}
+            </AuLabel>
+            <PowerSelect
+              @allowClear={{true}}
+              @searchEnabled={{true}}
+              @noMatchesMessage={{t "roadSignConcept.crud.noData"}}
+              @searchField="label"
+              @options={{@model.categories}}
+              @selected={{this.category}}
+              @onChange={{this.handleCategorySelection}} as |category|
```
Comment (Windvis) [inline app/templates/road-sign-concepts/road-sign-concept.hbs:436]: Should this use the same dataset as the related sidebar? This action does some filtering which might need to be adjusted for this?

**Review by Windvis — CHANGES_REQUESTED** [review]

File: app/routes/road-sign-concepts/road-sign-concept.js (hunk 2 above)
Comment (bdevloed) [reply app/routes/road-sign-concepts/road-sign-concept.js:21]: No :-)

**Comment by Windvis** [comment]
@bdevloed For the sidebar to display correctly you need to add your new flag to this getter:
https://github.com/lblod/frontend-mow-registry/blob/dc9d5436562e8a4d1b8cccfa23e84f70e23b8d80/app/controllers/road-sign-concepts/road-sign-concept.js#L19

File: app/templates/road-sign-concepts/road-sign-concept.hbs (hunk 4 above)
Comment (bdevloed) [reply app/templates/road-sign-concepts/road-sign-concept.hbs:436]: The thinking behind this is that a "main" and "related" are mutually exclusive. I've adapted the filter now to reflect this.

**Pull request merged**